src/Programs.tsx

```diff
@@ -0,0 +1,326 @@
+import React, { ReactElement, useState, SetStateAction, Dispatch, useEffect, useContext } from 'react';
+import { Button, Table, Header, Grid, Segment } from 'semantic-ui-react';
+import { setUserProgram, retrievePrograms } from './Database';
+import { NumericOnlyInput } from './components';
+import { NewProgramContext } from './Context';
+
+export interface Program {
+  name: string;
+  author: string;
+  level: string;
+  days: string;
+  program: Array<Array<{ exerciseName: string; Sets: number; Reps: number }>>;
+  notes: string;
+  setViewedName: Dispatch<SetStateAction<string>>;
+  setEditing: Dispatch<SetStateAction<boolean>>;
+}
+
+type ProgramNoMethods = Omit<Program, 'setViewedName' | 'setEditing'>;
+
+function Programs(): ReactElement {
+  const [viewedName, setViewedName] = useState<string>('');
+  const [editing, setEditing] = useState<boolean>(false);
+  const [database] = useState<Array<ProgramNoMethods>>(retrievePrograms());
+  const [rows] = useState<Array<ReactElement>>(() => {
+    const temp: Array<ReactElement> = [];
+    for (let i = 0; i < database.length; i++) {
+      temp.push(
+        <Grid.Row>
+          <Row {...database[i]} setViewedName={setViewedName} />
+        </Grid.Row>,
+      );
+    }
+    return temp;
+  });
+
+  if (editing) {
+    return <Edit {...(database.find((el) => el.name === viewedName) as ProgramNoMethods)} setEditing={setEditing} />;
+  }
+  if (viewedName) {
+    return <View {...(retrievePrograms().find((el) => el.name === viewedName) as ProgramNoMethods)} setViewedName={setViewedName} setEditing={setEditing} />;
+  }
+  return <Grid divided="vertically">{rows}</Grid>;
```

Comment:
Instead of having `rows`  you should just do a database.map() to return the rows, that's how you should use React.
```
return <Grid divided="vertically">{database.map(item => <Grid.Row>
          <Row {...item} setViewedName={setViewedName} />
        </Grid.Row>)}</Grid>;
```

Reply:
fixed